packages/@aws-cdk/alexa-ask/lib/pipeline-actions.ts

```diff
@@ -0,0 +1,69 @@
+import codepipeline = require('@aws-cdk/aws-codepipeline-api');
+import cdk = require('@aws-cdk/cdk');
+
+/**
+ * Construction properties of the {@link AlexaSkillDeployAction Alexa deploy Action}.
+ */
+export interface AlexaSkillDeployActionProps extends codepipeline.CommonActionProps,
+    codepipeline.CommonActionConstructProps {
+
+  /**
+   * The client id of the developer console token
+   */
+  clientId: cdk.Secret;
+
+  /**
+   * The client secret of the developer console token
+   */
+  clientSecret: cdk.Secret;
+
+  /**
+   * The refresh token of the developer console token
+   */
+  refreshToken: cdk.Secret;
+
+  /**
+   * The Alexa skill id
+   */
+  skillId: string;
+
+  /**
+   * The source artifact containing the voice model and skill manifest
+   */
+  inputArtifact: codepipeline.Artifact;
+
+  /**
+   * An optional artifact containing overrides for the skill manifest
+   */
+  overrideArtifact?: codepipeline.Artifact;
+}
+
+/**
+ * Deploys the skill to Alexa
+ */
+export class AlexaSkillDeployAction extends codepipeline.DeployAction {
+  constructor(scope: cdk.Construct, id: string, props: AlexaSkillDeployActionProps) {
+    super(scope, id, {
+      stage: props.stage,
+      runOrder: props.runOrder,
+      artifactBounds: {
+        minInputs: 1,
+        maxInputs: 2,
+        minOutputs: 0,
+        maxOutputs: 1,
```

Comment:
Can this Action really have outputs? Deployment Actions almost never do... If 'yes', what is that output?